### PR TITLE
[Search] Percentile, license and mobile friendly search filters

### PR DESCRIPTION
**Pull request by yattias**
- Adding license as a filter
- Adding percentile as a filter
- Using `<SwipableDrawer />` and `<Modal>` components to display search filters in smaller resolutions
- While at it, converted `SearchResultsForDocs` component into typescript and extracted filtering logic into `<SearchFiltersForDocs />`  and `SearchFiltersContext`
- Fix back button issue redirecting to wrong place
image

image

image

image

### CODE REVIEW

File: components/Search/SearchFiltersForDocs.tsx
Comment (yattias) [inline components/Search/SearchFiltersForDocs.tsx:1]: This is a new component that was extracted from mega component of `SearchResultsForDocs`

File: components/Search/lib/AppliedFilters.tsx
Comment (yattias) [inline components/Search/lib/AppliedFilters.tsx:1]: New component that was extracted from `SearchResultsForDocs` it holds rendering logic for applied filters

File: components/Search/lib/SearchFiltersContext.tsx
Comment (yattias) [inline components/Search/lib/SearchFiltersContext.tsx:1]: The purpose of this context is to hold filtering state that can be used across multiple components

File: components/Form/SimpleSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    setValue(newValue as number);
+    onChange(newValue as number);
+  };
+
+  const handleSliderChange = (event: Event, newValue: number | number[]) => {
+    setValue(newValue as number);
+  };
+
+  function valueLabelFormat(value: number) {
+    return value + "th percentile";
```
Comment (koutst) [inline components/Form/SimpleSlider.tsx:25]: Nit: This won't work for every percentile. 1st, 2nd, 3rd, 21st, 23rd, 24th etc.

**Review by koutst — APPROVED** [review]
LGTM

File: components/Form/SimpleSlider.tsx (hunk 1 above)
Comment (yattias) [reply components/Form/SimpleSlider.tsx:25]: good call

**Pull request merged**